src/test/java/com/iota/iri/network/neighbor/impl/NeighborImplTest.java

```diff
@@ -207,4 +209,62 @@ public void markingTheNeighborForDisconnectWillNeverMakeItReadyForMessagesAgain(
         neighbor.setState(NeighborState.READY_FOR_MESSAGES);
         assertEquals("should be marked for disconnect", NeighborState.MARKED_FOR_DISCONNECT, neighbor.getState());
     }
+
+    @Test
+    public void writeHeartbeat() {
+        Heartbeat heartbeat = new Heartbeat();
+        heartbeat.setFirstSolidMilestoneIndex(1);
+        heartbeat.setLastSolidMilestoneIndex(2);
+        ByteBuffer heartbeatPacket =  Protocol.createHeartbeatPacket(heartbeat);
+
+        Neighbor neighbor = new NeighborImpl<>(selector, new FakeChannel() {
+
+            @Override
+            public int write(ByteBuffer buf) {
+                int bytesWritten = 0;
+                while (buf.hasRemaining()) {
+                    buf.get();
+                    bytesWritten++;
+                }
+                return bytesWritten;
+            }
+        }, localAddr, serverSocketPort, pipeline);
+
+        neighbor.send(heartbeatPacket);
+
+        try {
+            assertEquals("should have written the entire packet", heartbeatPacket.capacity(), neighbor.write());
+        } catch (IOException e) {
+            fail("didn't expect an exception");
```

Comment:
This needs a better description :D

Reply:
agree, at least print the exception message